src/targets/gpu/mlir_conv.cpp

```diff
@@ -0,0 +1,291 @@
+            {
+                migraphx::value::binary bin(bin_size);
+                if(miirBufferGet(handle.get(), reinterpret_cast<char*>(bin.data()), &bin_size) ==
+                   MIIR_SUCCESS)
+                {
+                    size_t global_size, block_size;
+                    if(miirGetExecutionDims(handle.get(), &global_size, &block_size) ==
+                       MIIR_SUCCESS)
+                    {
+                        printf("MLIR - %s\n", mlir_options.c_str());
+                        result = std::make_shared<execution_spec>(
+                            std::move(bin), global_size, block_size);
+                    }
+                }
+            }
+
+            binary_map[mlir_options] = result;
+        }
+        else
+        {
+            result = bin_i->second;
+        }
+#endif // MIGRAPHX_MLIR_MIOPEN_SUPPORT
+        return result;
+    }
+
+    instruction_ref get_literal(uint64_t value)
+    {
+        auto fi = literal_map.find(value);
+        if(fi != literal_map.end())
+            return fi->second;
+        auto lit = prog->add_literal(value);
+        literal_map.emplace(value, lit);
+        return lit;
+    }
+
+    operation make_code_object_op(instruction_ref op_r, std::shared_ptr<execution_spec> spec)
+    {
+        // each pointer is expanded out to a MemRefDescriptor
+        auto inp_t = op_r->inputs().at(0)->get_shape();
+        auto flt_t = op_r->inputs().at(1)->get_shape();
+        auto out_t = op_r->get_shape();
+
+        auto i64 = shape(shape::uint64_type);
+
+        std::vector<shape> expected_inputs = {
+            flt_t, flt_t, i64, i64, i64, i64, i64, i64, i64, i64, i64, i64,  i64,   inp_t,
+            inp_t, i64,   i64, i64, i64, i64, i64, i64, i64, i64, i64, i64,  out_t, out_t,
+            i64,   i64,   i64, i64, i64, i64, i64, i64, i64, i64, i64, out_t};
+
+        return migraphx::make_op("gpu::code_object",
+                                 {
+                                     {"code_object", spec->binary},
+                                     {"symbol_name", mlir_kernel_name},
+                                     {"global", spec->global_size},
+                                     {"local", spec->local_size},
+                                     {"expected_inputs", migraphx::to_value(expected_inputs)},
+                                     {"output", migraphx::to_value(out_t)},
+                                 });
+    }
+
+    void add_memref_descriptor(std::vector<instruction_ref>& refs, instruction_ref inst)
+    {
+        const size_t offset = 0;
+        auto inst_t         = inst->get_shape();
+        refs.push_back(inst);
+        refs.push_back(inst);
+        refs.push_back(get_literal(offset)); // offset
+
+        // dim sizes
+        std::for_each(inst_t.lens().begin(), inst_t.lens().end(), [this, &refs](const auto lval) {
+            refs.push_back(get_literal(lval));
+        });
```

Comment:
This can be done with `std::transform` using `std::back_inserter`:

```cpp
std::transform(inst_t.lens().begin(), inst_t.lens().end(), std::back_inserter(refs), [&](const auto& lval) {
    return get_literal(lval);
});
```

Reply:
Cool, I will try that.